src/topojsonm/topojsonm.ml

```diff
@@ -1 +1,227 @@
+        arr so_far stack
+    | Stack.In_object (n, ms, stack) ->
+        let so_far = (n, v) :: ms in
+        obj so_far stack
+    | Stack.Empty -> v
+  in
+  enter (dec ()) Empty
+
+let encode_value e json =
+  let module Stack = struct
+    type t =
+      | In_array of Ezjsonm.value list * t
+      | In_object of (string * Ezjsonm.value) list * t
+      | Empty
+  end in
+  let enc e l = ignore (Jsonm.encode e (`Lexeme l)) in
+  let rec t v e stack =
+    match v with
+    | `A vs ->
+        enc e `As;
+        arr vs e stack
+    | `O ms ->
+        enc e `Os;
+        obj ms e stack
+  and value v e stack =
+    match v with
+    | (`Null | `Bool _ | `Float _ | `String _) as v ->
+        enc e v;
+        continue e stack
+    | #Ezjsonm.t as x -> t (x :> Ezjsonm.t) e stack
+  and arr vs e stack =
+    match vs with
+    | v :: vs' ->
+        let stack = Stack.In_array (vs', stack) in
+        value v e stack
+    | [] ->
+        enc e `Ae;
+        continue e stack
+  and obj ms e stack =
+    match ms with
+    | (n, v) :: ms ->
+        enc e (`Name n);
+        let stack = Stack.In_object (ms, stack) in
+        value v e stack
+    | [] ->
+        enc e `Oe;
+        continue e stack
+  and continue e stack =
+    match stack with
+    | Stack.In_array (vs, stack) -> arr vs e stack
+    | Stack.In_object (ms, stack) -> obj ms e stack
+    | Stack.Empty -> ()
+  in
+  value json e Stack.Empty
+
+let map_object f src dst =
+  let decoder = Jsonm.decoder src in
+  let encoder = Jsonm.encoder dst in
+  let loc () = Jsonm.decoded_range decoder in
+  let enc v =
+    match Jsonm.encode encoder v with
+    | `Ok -> ()
+    | `Partial -> raise (Abort (`Unexpected "partial encoding"))
+  in
+  let rec go () =
+    match Jsonm.decode decoder with
+    | `Lexeme (`Name "object" as t) ->
+        (match Jsonm.decode decoder with
+        | `Lexeme (`Name "example" as ex) -> (
+            match Topo.of_json @@ decode_single_object decoder with
+            | Error (`Msg m) -> raise (Abort (`Unexpected m))
+            | Ok v -> (
+                match Topo.topojson v with
+                | Geometry g ->
+                    let g' = f g in
+                    enc (`Lexeme ex);
+                    encode_value encoder
+                      (Topo.to_json @@ Topo.v ?bbox:(Topo.bbox v) (Geometry g'));
+                    go ()
+                | _ -> raise (Invalid_argument "Expected a geometry object")))
+        | `Lexeme _ as t ->
+            enc t;
+            go ()
+        | `Error e -> raise (Abort (`Error (loc (), e)))
+        | `End -> ignore @@ Jsonm.encode encoder `End
+        | `Await -> assert false);
+        enc (`Lexeme t)
+    | `Lexeme _ as t ->
+        enc t;
+        go ()
+    | `Error e -> raise (Abort (`Error (loc (), e)))
+    | `End -> ignore @@ Jsonm.encode encoder `End
+    | `Await -> assert false
+  in
+  try Ok (go ()) with Abort e -> Error e
+
+let map_props f src dst =
```

Comment:
Let's remove any imported code from Geojson that we don't need, otherwise the PR is quite big and hard to review.